src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -596,6 +648,30 @@ public SuiteResult getSuite(String name) {
         return suitesByName.get(name);
     }
 
+    @Nonnull
+    public TestResult getResultByRunAndNode(@Nonnull String runId, @Nonnull String nodeId) {
+        return getResultByRunAndNodes(runId, Collections.singletonList(nodeId));
+    }
+
+    @Nonnull
+    public TestResult getResultByRunAndNodes(@Nonnull String runId, @Nonnull List<String> nodeIds) {
+        TestResult result = new TestResult();
+        if (suitesByRunAndNode.containsKey(runId)) {
+            for (String n : nodeIds) {
+                List<SuiteResult> suites = suitesByRunAndNode.get(runId).get(n);
+                if (suites != null) {
+                    for (SuiteResult s : suites) {
+                        s.setParent(null);
```

Comment:
I'm not entirely clear what the implications of setParent are here -- it feels rather odd to see this in the getter though, since isn't it changing the persistent state?